modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -624,6 +624,142 @@ struct InferList: public cv::detail::KernelTag {
     }
 };
 
+struct InferList2: public cv::detail::KernelTag {
+    using API = cv::GInferList2Base;
+    static cv::gapi::GBackend backend()  { return cv::gapi::ie::backend(); }
+    static KImpl kernel()                { return KImpl{outMeta, run}; }
+
+    static cv::GMetaArgs outMeta(const ade::Graph      &gr,
+                                 const ade::NodeHandle &nh,
+                                 const cv::GMetaArgs   &in_metas,
+                                 const cv::GArgs       &/*in_args*/) {
+        // Specify the input information to the IE from the framework
+        // NB: Have no clue if network's input [dimensions] may ever define
+        // its output dimensions. It seems possible with OpenCV DNN APIs
+
+        GConstGIEModel gm(gr);
+        const auto &uu = gm.metadata(nh).get<IEUnit>();
+
+        // Initialize input information
+        // Note our input layers list order matches the API order and so
+        // meta order.
+        GAPI_Assert(uu.params.input_names.size() == (in_metas.size() - 1u)
+                    && "Known input layers count doesn't match input meta count");
+
+        // In contrast to InferList, the InferList2 has only one
+        // "full-frame" image argument, and all the rest are arrays of
+        // ether ROI or blobs. So here we set the 0th arg image format
+        // to all inputs which are ROI-based (skipping the
+        // "blob"-based ones)
+        // FIXME: this is filtering not done, actually! GArrayDesc has
+        // no hint for type!
+        const auto &mm_0   = in_metas[0u];
+        const auto &meta_0 = util::get<cv::GMatDesc>(mm_0);
+        GAPI_Assert(   !meta_0.isND()
+                    && !meta_0.planar
+                    && "Only images are supported as the 0th argument");
+        std::size_t idx = 1u;
+        for (auto &&input_name : uu.params.input_names) {
+                  auto &ii = uu.inputs.at(input_name);
+            const auto &mm = in_metas[idx++];
+            GAPI_Assert(util::holds_alternative<cv::GArrayDesc>(mm)
+                        && "Non-array inputs are not supported");
+
+            if (ii->getTensorDesc().getDims().size() != 2) {
+                // FIXME: SUPER-MEGA-KLUDGE!!!
+                ii->setPrecision(toIE(meta_0.depth));
+                ii->setLayout(IE::Layout::NHWC);
```

Comment:
This is a misuse of the API, btw

Reply:
(here we set a network input layout what means for IE "compile for this layout", but it makes little sense since then we call `SetBlob`)